packages/grid/_modules_/grid/hooks/features/clipboard/useGridClipboard.ts

```diff
@@ -55,7 +55,12 @@ export const useGridClipboard = (apiRef: GridApiRef): void => {
   const handleKeydown = React.useCallback(
     (event: KeyboardEvent) => {
       const isModifierKeyPressed = event.ctrlKey || event.metaKey || event.altKey;
-      if (event.key.toLowerCase() !== 'c' || !isModifierKeyPressed) {
+      /**
+       * On some systems (e.g. MacOS) Alt+C might add accents to letters (like Alt+c gives ç)
+       * depending on keyboard layout and language.
+       * In this case `event.code` is used to check the actual key pressed.
+       */
+      if ((event.key.toLowerCase() !== 'c' && event.code !== 'KeyC') || !isModifierKeyPressed) {
```

Comment:
Would it be enough to check the physically keyboard only?

```suggestion
      if (event.code !== 'KeyC' || !isModifierKeyPressed) {
```

Reply:
Well, I'm just being cautious here.

There are different layouts, and for example `KeyC` will be returned for `C` key on QWERTY and `J` key on Dvorak.
If we remove `event.key`, `Alt+C` on Dvorak layout won't work, which is a bad UX.
So I think it's better to keep both.

Here's some context on this https://ux.stackexchange.com/questions/30666/keyboard-shortcuts-on-non-qwerty-keyboard-layouts

Reply:
The Dvorak keyboard case is interesting, I didn't know that it would have an influence it, it's the time I see it relevant in a discussion on MUI 😁. But for a macOS user with a Dvorak keyboard, would it mean that he would need to press Option + J? 

In this case maybe?

```suggestion
      // event.key === 'c' is not enough as alt+c can lead to © or other characters on macOS.
      // event.code === 'KeyC' is not enough as event.code assume a QWERTY keyboard layout which would be wrong with a Dvorak keyboard (as if pressing J).
      if ((event.keyCode !== 67 || !isModifierKeyPressed) {
```

Reply:
Indeed, that's the only solution that would work consistently on every layout (`Alt`/`Option` + `C`, where `C` is the key that is marked as `C` on system keyboard layout)

The problem with `keyCode` is that [it's deprecated](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) in favor of `key` and `code`. 
`keyCode` is still supported by browsers, since a lot of apps use it.

---

This made me think if `Alt`/`Option` + `C` is a good idea for keyboard shortcut. Because it has direct impact on another feature of `DataGrid` - Editing.

### The problem

When user presses any printable key, grid cell enters edit mode (as per https://mui.com/components/data-grid/editing/#start-editing)
As we have seen here, on MacOS `Option` is a modifier key, and `Option`+`C` on most layouts results in **printable key** (e.g. `ç` on British/U.S. layout) that IMHO should start edit mode. Currently, it doesn't and it's something we should fix.

### Proposal

Since `Alt+C` is the only shortcut that uses `Alt` (see https://mui.com/components/data-grid/accessibility/#keyboard-navigation) and it's not a widespread shortcut (as opposed to `Ctrl+C/Ctrl+V`), I propose to change it to something that doesn't involve `Alt` key:

`Ctrl+Shift+C`

I think it makes more sense.
We need to add `event.preventDefault()` though, since this shortcut opens DevTools on MacOS 🙃

### Doubts

- **Should we consider this a breaking change?**
Technically it's a breaking change, but the functionality doesn't work on MacOS.
We can leave current implementation for backward compatibility though.
What do you think?

### Action points

- change `Alt+C` shortcut to `Ctrl+Shift+C` 
- start editing cell when user presses printable key modified with `Alt`/`Option` - this might be a tricky one, since I'm not sure yet how we define "printable key". This can be explored in a separate issue.

Would love some input from @mui-org/x members

Reply:
+1 to fix quick and dirty in v5, and cover in https://github.com/mui-org/material-ui-x/issues/3287 that we might be able to improve this with a breaking change in v6.